### PR TITLE
add pre-commit hooks

### PR DESCRIPTION
**Pull request by profjsb**
add details for using pre-commit to contributing doc

### CODE REVIEW

File: contributing.MD [hunk 1]
```diff
@@ -11,7 +11,21 @@ the `upstream` remote:
...
+```bash
+pre-commit install
+```
+
+This will only need to be done once. After that, all your commits will be checked for code formatting issues.
+
+Make a local branch like:
+
+```bash
+git checkout -b your_local_branch_with_changes_name
```
Comment (stefanv) [inline contributing.MD:25]:
```suggestion
git switch -c your_local_branch_with_changes_name
```

File: pyproject.toml [hunk 2]
```diff
@@ -25,7 +25,10 @@ dependencies = [
     "joblib",
     "pytest",
     "tqdm",
-    "multiprocess"
+    "multiprocess",
+    "wandb",
```
Comment (stefanv) [inline pyproject.toml:29]: This is a package dependency?

**Review by stefanv — COMMENTED** [review]
Thanks, Josh, this looks good.

I would replace flake8 with ruff, but that can also be done in another PR.

See, e.g., https://github.com/networkx/networkx/blob/main/.pre-commit-config.yaml#L19

and an example ruff configuration

https://github.com/networkx/networkx/blob/main/pyproject.toml#L169

File: pyproject.toml (hunk 2 above)
Comment (profjsb) [reply pyproject.toml:29]: yes

**Comment by stefanv** [comment]
I've never seen a capitalized `contributing.MD` before, but as long as GH recognizes it probably OK.

**Comment by stefanv** [comment]
[GH docs](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) say it's case insensitive :+1:

**Comment by stefanv** [comment]
Shall we merge this and I can add ruff in another PR?

File: pyproject.toml (hunk 2 above)
Comment (kmzzhang) [reply pyproject.toml:29]: Wandb is only required when wandb=True is specified. Is it preferred to add it as a project dependency in this case?

**Comment by stefanv** [comment]
See https://github.com/kmzzhang/nbi/pull/25

**Pull request merged**